libcsl_godot/src/lib.rs

```diff
@@ -329,10 +435,32 @@ impl GTxBuilder {
             utxos.add(&gutxo.bind().to_transaction_unspent_output());
         });
         let mut tx_builder = self.tx_builder.clone();
-        tx_builder.set_inputs(&self.inputs_builder);
+        tx_builder.set_inputs(&self.inputs_builder.clone());
         tx_builder
             .add_inputs_from(&utxos, CoinSelectionStrategyCIP2::LargestFirstMultiAsset)
             .expect("Could not add inputs");
+
+        tx_builder.set_mint_builder(&self.mint_builder.clone());
+        if self.redeemers.len() > 0 {
+            let min_collateral = self.fee * 150 / 100 + 1;
+            let collateral_amount = Gd::from_object(
+                BigInt::from_int(min_collateral.try_into().unwrap())
+            );
+            for gutxo in gutxos.iter_shared() {
```

Comment:
I am not sure I follow what is going on here... Only if there are redeemers in the transaction (meaning that we mint something), for each of the UTxOs passed to the function, we set a collateral _and_ total_collateral of the transaction (as well as the change address). Why is this done for each UTxO? Why only when there are redeemers in the transaction?